packages/web-components/src/button/button.ts

```diff
@@ -27,6 +28,18 @@ export class Button extends FASTElement {
   @attr
   public appearance?: ButtonAppearance;
 
+  /**
+   * Handles changes to appearance attribute custom states
+   * @param prev - the previous state
+   * @param next - the next state
+   */
+  public appearanceChanged(prev: ButtonAppearance | undefined, next: ButtonAppearance | undefined) {
+    if (prev) {
+      toggleState(this.elementInternals, `${prev}`, false);
+    }
+    toggleState(this.elementInternals, `${next}`, true);
```

Comment:
You mentioned this offline, but it would be good to eventually have an abstraction here for toggling states like this if we're going to use this pattern more. Like `toggleBooleanState(prev, next)` or something like that.

Reply:
Oh, I like that as an option - I definitely think much of the duplication can be reduced once we get "states" in all-up and this would be a good ergonomic enhancement.